test/java/org/chromium/net/CertificateVerificationTest.java

```diff
@@ -0,0 +1,72 @@
+package org.chromium.net;
+
+import static org.junit.Assert.assertEquals;
+
+import io.envoyproxy.envoymobile.engine.AndroidJniLibrary;
+import io.envoyproxy.envoymobile.engine.JniLibrary;
+
+import org.junit.After;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.robolectric.RobolectricTestRunner;
+
+/**
+ * Simple test for Certificate verification.
+ */
+@RunWith(RobolectricTestRunner.class)
+public class CertificateVerificationTest {
+  static {
+    AndroidJniLibrary.loadTestLibrary();
+    JniLibrary.load();
+  }
+
+  @Before
+  public void setUp() throws Exception {
+    AndroidNetworkLibrary.setFakeCertificateVerificationForTesting(true);
+  }
+
+  @After
+  public void tearDown() throws Exception {
+    JniLibrary.callClearTestRootCertificateFromNative();
+    AndroidNetworkLibrary.setFakeCertificateVerificationForTesting(true);
+  }
+
+  @Test
+  public void testVerifyCertificateCall() throws Exception {
```

Comment:
nit: there are a number of different cases being tested in this method. (for example no trusted roots are provide, cases where the certs match vs mismatch.) It would be nice to break this into a number of different tests so that if there are failures, the name of the failed test explains the incorrect behavior.